File: 异常-捕获/error_test.py

try:
    raise TypeError("类型错误");
    open(".../xxxxxx.ts")
except IOError as ioe:
    print("io error", ioe)
except TypeError as te:
    print("type error", te);
except Exception as e:
    print("common error", e.args)
finally:
    print("执行完毕")

File: 异常-捕获/error_test.js
const fs = require("fs");

const print = console.log;
try {

    throw new TypeError("无效的类型")
    fs.openSync("../sdasdas/x.ts")

} catch (err) {
    if (err instanceof TypeError) {
        print("type error", err)
    } else if (err instanceof Error) {
        print("common error", err)
    }
}

